gcp.go

```diff
@@ -144,15 +143,73 @@ func updateConfigWithGCPSettings(c *gwconfig.Config) error {
 		gcpMetadata[key] = value
 	}
 	c.WorkerTypeMetadata["gcp"] = gcpMetadata
-	c.WorkerID = gcpMetadata["id"].(string)
+	c.WorkerID = "gcp-" + gcpMetadata["id"].(string)
```

Comment:
If the `workerId` isn't going to match the gcp instance id, can the worker manager decide the workerId, rather than the worker, and pass it to the worker? It feels like if the worker manager names the worker, it will be easier to build a consistent view across the client and services, otherwise perhaps it will be tricky for the worker manager to preempt what the worker will choose for its `workerId`?

Reply:
Make sense to me!

Reply:
Oh yeah, I remember why I didn't do this in the first place. It would mean that we give up on the thing where the `instanceId` is at least knowable given a `workerId` since we don't know `instanceId` until after we make the request to spawn the worker. So is a `workerId` of `gcp-${slugid.nice()}` good enough for us? We can keep some data in worker-manager about what the `instanceId` is for later use in that case?

Reply:
I think the issue you describe is only a problem for EC2?  In that case, maybe a slugId is the best we can do.  But in either EC2 or GCP, I think it makes sense for the worker-manager to tell the worker what its workerId is as @petemoore is suggesting.  I don't see how that would mean (in the GCP case) we can't still use instanceId = workerId?

Reply:
We don't know `instanceId` at the time that we set the metadata that generic-worker uses to configure itself. I think I'm misunderstanding something.

Reply:
I was thinking of returning the workerId from the call to the credentialsUrl.  So *worker-manager* knows the workerId once the create-new-instance call returns, but the worker itself only finds out when it gets its credentials.  I'm probably the one missing something, though :)